rustworkx-core/src/connectivity/cycle_basis.rs

```diff
@@ -29,32 +31,45 @@ use std::hash::Hash;
 /// It may produce incorrect/unexpected results if the input graph has
 /// parallel edges.
 ///
-///
 /// Arguments:
 ///
 /// * `graph` - The graph in which to find the basis.
 /// * `root` - Optional node index for starting the basis search. If not
 ///     specified, an arbitrary node is chosen.
-///
-/// # Example
-/// ```rust
-/// use petgraph::prelude::*;
-/// use rustworkx_core::connectivity::cycle_basis;
-///
-/// let edge_list = [(0, 1), (0, 3), (0, 5), (1, 2), (2, 3), (3, 4), (4, 5)];
-/// let graph = UnGraph::<i32, i32>::from_edges(&edge_list);
-/// let mut res: Vec<Vec<NodeIndex>> = cycle_basis(&graph, Some(NodeIndex::new(0)));
-/// ```
-pub fn cycle_basis<G>(graph: G, root: Option<G::NodeId>) -> Vec<Vec<G::NodeId>>
+/// * `edges` - bool for when the user requests the edges instead
+///     of the nodes of the cycles.
+
+fn inner_cycle_basis<G>(
+    graph: G,
+    root: Option<G::NodeId>,
+    edges: bool,
+) -> EdgesOrNodes<G::NodeId, G::EdgeId>
 where
     G: NodeCount,
     G: IntoNeighbors,
+    G: IntoEdges,
     G: IntoNodeIdentifiers,
     G::NodeId: Eq + Hash,
+    G::EdgeId: Eq + Hash,
 {
     let mut root_node = root;
     let mut graph_nodes: HashSet<G::NodeId> = graph.node_identifiers().collect();
-    let mut cycles: Vec<Vec<G::NodeId>> = Vec::new();
+    let mut cycles_edges: Vec<Vec<G::EdgeId>> = Vec::new();
+    let mut cycles_nodes: Vec<Vec<G::NodeId>> = Vec::new();
+
+    // Method used to retrieve all the edges between an origin node and a target node.
```

Comment:
It's technically a docstring even though it won't get rendered in any doc builds.

```suggestion
    /// Method used to retrieve all the edges between an origin node and a target node.
```